dash/dash.py

```diff
@@ -1928,7 +1929,12 @@ def run_server(
         If a parameter can be set by an environment variable, that is listed
         too. Values provided here take precedence over environment variables.
 
+        :param host: Host IP used to serve the application
+            env: ``HOST``
+        :type host: string
+
         :param port: Port used to serve the application
+            env: ``PORT``
         :type port: int
```

Comment:
Just mark `port` here as either an int or a string containing an int, and we should be covered re: data types. Your coercion and try/except below looks great!

Can you please add a couple of tests of the failure cases, and maybe one of setting an integer string port and verifying that an app shows up there? These can probably all go in https://github.com/plotly/dash/blob/dev/tests/unit/test_configs.py

The failures can just be something like:
```py
app = Dash()
app.layout = html.Div()
with pytest.raises(...) as excinfo:
    app.run_server(port="garbage")
assert excinfo.value == ...
```

And the pass case can be something like:
```py
app = Dash()
app.layout = html.Div("hi", id="out")
dash_duo.run_server(app, port="12345")
assert dash_duo.server_url == "https://127.0.0.1:12345"
dash_duo.wait_for_text_to_equal("out", "hi")
```